flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -271,7 +269,14 @@ protected StreamTask(
 			new ExecutorThreadFactory("AsyncOperations", uncaughtExceptionHandler));
 
 		this.stateBackend = createStateBackend();
-		this.checkpointStorage = stateBackend.createCheckpointStorage(getEnvironment().getJobID());
+
+		this.subtaskCheckpointCoordinator = new SubtaskCheckpointCoordinatorImpl(
+			stateBackend.createCheckpointStorage(getEnvironment().getJobID()),
```

Comment:
nit: we can get `environment` directly from argument instead of `getEnvironment()`, also for `getAsyncOperationsThreadPool()`

Reply:
replied in the comment above 
(this would bring inconsistency with values from getters)